### ISSUE TITLE
BC scenarization : BindingConstraintsRepository::enabled() function

### ISSUE THREAD

**Issue by guilpier-code** [AntaresSimulatorTeam/Antares_Simulator#1387]
**BindingConstraintsRepository::enabled()** function raises a problem.
Currently, among all existing binding constraints, it selects and return the enabled ones. 
But this selection is made only it hasn't already been done.
This function is called in many places in the Antares **solver**, but is not called from the UI code base.
This function is more complex than it should be  : as it is only called from the **solver**, there is no risk that the list of enabled BC changes throughout the **solver** execution. 
So selecting enabled BCs should be done right after reading the binding constraints input data, and once for all.
The **enabled()** function should only return an already filled container.

To go further, as suggested @JasonMarechal, why do we keep a list of existing BCs (enabled or disabled) in the **solver** code base while we could only use a list of enabled BCs ? Disabled BCs are useless in the context of the **solver**.

**To sum up**, the solver code base should : 
- Maintain a container of enabled BCs only (not interested with disabled BCs)
- Select and store enabled BCs once, and right after loading input data.
- Make **enabled()** function only return the container of enabled BCs, and this function should not do anything else.

**Comment by flomnes** [comment]
I agree. It would simplify the code to only load enabled binding constraints, and ignore the rest.

However, the GUI still requires loading disabled binding constraints. Furthermore, we should introduce genericity for properties such as
- enabled/disabled
- name

to avoid re-coding the associated behaviour for each new objetc.

**Comment by flomnes** [comment]
@guilpier-code, there is now 
```
std::vector<std::shared_ptr<BindingConstraint>> BindingConstraintsRepository::activeContraints() const
```
This should be enough.
